### PR TITLE
Houdini: Update ayon_lop_import 1.1

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
update ayon_lop_import 1.1: Allow users to select folder path and products using some actions

Animation_45

### CODE REVIEW

**Comment by MustafaJafar** [comment]
Also, I think Representation parm should be renamed to `Product Type`. 
image

**Comment by MustafaJafar** [comment]
Could you add you as a reviewer ? because I'm not able to do it @BigRoy

File: client/ayon_core/hosts/houdini/api/hda_utils.py [hunk 1]
```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
...
+
+    )
+
+    items = list(map(lambda p: p["name"], products))
+
+    result = []
+
+    for i in items:
+        result.append(i)
+        result.append(i)
```
Comment (BigRoy) [inline client/ayon_core/hosts/houdini/api/hda_utils.py]:
```suggestion

    items = list(map(lambda p: p["name"], products))

    result = []

    for i in items:
        result.append(i)
        result.append(i)
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py [hunk 2]
```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
...
+                                                folder_path,
+                                                fields=id_only)
+    my_folder_id = folder_entity["id"]
+
+    products = ayon_api.get_products(
+        project_name,
+        folder_ids = [my_folder_id],
+        product_types = [my_product_type]
+
+    )
```
Comment (BigRoy) [inline client/ayon_core/hosts/houdini/api/hda_utils.py]:
```suggestion
    folder_entity = ayon_api.get_folder_by_path(project_name,
                                                folder_path,
                                                fields={"id"})
    if not folder_entity:
        return []

    products = ayon_api.get_products(
        project_name,
        folder_ids=[folder_entity["id"]],
        product_types=[my_product_type]
    )
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py [hunk 3]
```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
...
+        This function should be refactored.
+        It currently shows shows the available
+          folder paths within the current project only.
+        It also should include a button to set the the folder path
+          parameter to `$AYON_FOLDER_PATH`.
+    """
+    from ayon_core.tools.publisher.widgets.folders_dialog import FoldersDialog
+    from ayon_core.tools.utils.host_tools import get_tool_by_name
+    from ayon_core.hosts.houdini.api.lib import get_main_window
+
```
Comment (BigRoy) [inline client/ayon_core/hosts/houdini/api/hda_utils.py:470]:
```suggestion
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py [hunk 4]
```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
...
+      within the specified folder path with in the specified project.
+
+    Users can specify those in the HDA parameters.
+
+    Returns:
+        List[str]: Product options for Products menu.
+    """
+    project_name = node.evalParm("project_name")
+    folder_path = node.evalParm("folder_path")
+    my_product_type = node.evalParm("representation_name")
```
Comment (BigRoy) [inline client/ayon_core/hosts/houdini/api/hda_utils.py:430]: Why is `representation_name` the `product_type` according to you? :) Seems like you're missing up terminology?

```suggestion
    representation_name = node.evalParm("representation_name")
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 2 above)
Comment (BigRoy) [reply client/ayon_core/hosts/houdini/api/hda_utils.py]: I think the product types filtering is incorrect maybe?

File: client/ayon_core/hosts/houdini/api/hda_utils.py [hunk 5]
```diff
@@ -412,3 +412,79 @@ def keep_background_images_linked(node, old_name):
 
     if changes:
         set_background_images(parent, images)
+
+
+def get_products_menu_items(node):
+    """get products menu items
```
Comment (BigRoy) [inline client/ayon_core/hosts/houdini/api/hda_utils.py]:
```suggestion
    """Return products menu items
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 1 above)
Comment (MustafaJafar) [reply client/ayon_core/hosts/houdini/api/hda_utils.py]: I can't tell the difference.

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 2 above)
Comment (MustafaJafar) [reply client/ayon_core/hosts/houdini/api/hda_utils.py]: I can't tell if it's incorrect.

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 4 above)
Comment (MustafaJafar) [reply client/ayon_core/hosts/houdini/api/hda_utils.py:430]: I think it's `product_type`, that's why I suggested changing the parameter label to Product Type

This what I found when I tested it. 
image

Also, the `product_type` for usd products is the same as their `representation` which is `usd`
image

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 1 above)
Comment (MustafaJafar) [reply client/ayon_core/hosts/houdini/api/hda_utils.py]: I think you meant: 
```suggestion

    result = []
    for product in products:
        result .append(product["name"])
        result .append(product["name"])
```

File: client/ayon_core/hosts/houdini/api/hda_utils.py (hunk 4 above)
Comment (BigRoy) [reply client/ayon_core/hosts/houdini/api/hda_utils.py:430]: > Also, the product_type for usd products is the same as their representation which is usd

This isn't required to be true. So let's stick to explicitly using the representation name.

Comment (MustafaJafar) [reply client/ayon_core/hosts/houdini/api/hda_utils.py:430]: > This isn't required to be true. So let's stick to explicitly using the representation name.

Ok, I'll change it back in the next commit.

But, My main concern is that some confusion may occur when adopting the same functionality with different products than `usd`

```python
project_name = "some project name"
folder_path = "some folder path"
representation_name = "pointcache"

folder_entity = ayon_api.get_folder_by_path(project_name,
                                                folder_path,
                                                fields={"id"})

products = ayon_api.get_products(
        project_name,
        folder_ids=[folder_entity["id"]],
        product_types=[representation_name]
    )
```

**Comment by MustafaJafar** [comment]
closed in favor of https://github.com/BigRoy/ayon-core/pull/9

**Pull request closed without merging**